src/Tizen.Sensor/Tizen.Sensor/Sensor.cs

```diff
@@ -545,5 +652,606 @@ private void DestroyHandles()
+        {
+            int error = Interop.SensorRecoder.RecorderDataGetDouble(data , key , out value);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error getting the sensor recorder double data");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderDataGetDouble Failed");
+            }
+        }
+
+        /// <summary>
+        /// Gets the start and the end time of the time period of a given record data.
+        /// </summary>
+        /// <param name="data">
+        /// Record data handle.
+        /// </param>
+        /// <param name="startTime">
+        /// Start time of the time period of the record.
+        /// </param>
+        /// <param name="endTime">
+        /// End time of the time period of the record.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderDataGetTime(IntPtr data, out IntPtr startTime, out IntPtr endTime)
+        {
+            int error = Interop.SensorRecoder.RecorderDataGetTime(data, out startTime, out endTime);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error getting the sensor recorder time");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderDataGetTime Failed");
+            }
+        }
+
+        /// <summary>
+        /// Destroys a recorder option handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderDestroyOption()
+        {
+            int error = Interop.SensorRecoder.RecorderDestroyOption(recorderOption);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error deleting the sensor recorder option");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderDestroyOption Failed");
+            }
+        }
+
+        /// <summary>
+        /// Destroys a recorder query handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderDestroyQuery()
+        {
+            int error = Interop.SensorRecoder.RecorderDestroyQuery(recorderQuery);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error deleting the sensor recorder query");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderDestroyQuery Failed");
+            }
+        }
+
+        /// <summary>
+        /// Checks whether it is supported to record a given sensor type.
+        /// </summary>
+        /// <param name="type">
+        /// A sensor type to check.
+        /// </param>
+        /// <param name="isSupported">
+        /// If supported.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderIsSupported(Enum type, out IntPtr isSupported)
```

Comment:
Do not use IntPtr in public API

Why do you use `System.Enum` class?

Reply:
Inptr is removed from all public API's and Enum is replaced with int